code/datums/extensions/cell/cell.dm

```diff
@@ -0,0 +1,163 @@
+	if(load_delay && !do_after(user, load_delay, holder))
+		return FALSE
+
+	// Recheck existing cell ref.
+	if(!istype(cell) || !istype(user) || QDELETED(cell) || QDELETED(user) || user.incapacitated())
+		return FALSE
+	// Recheck existing cell ref.
+	existing_cell = loaded_cell_ref?.resolve()
+	if(istype(existing_cell) && !QDELETED(existing_cell) && existing_cell.loc == holder)
+		to_chat(user, SPAN_WARNING("\The [holder] already has \a [existing_cell] loaded."))
+		return TRUE // technically a valid interaction.
+
+	// Load the cell.
+	if(user.try_unequip(cell, holder, FALSE))
+		user.visible_message(SPAN_NOTICE("\The [user] slots \the [cell] into \the [holder]."))
+		loaded_cell_ref = weakref(cell)
+		if(load_sound)
+			playsound(user.loc, pick(load_sound), 25, 1)
+		holder_item.update_icon()
+		return TRUE
+	return FALSE
+
+/datum/extension/loaded_cell/proc/try_unload(var/mob/user, var/obj/item/tool)
+
+	// Check inputs.
+	var/obj/item/holder_item = holder
+	if(!istype(user) || QDELETED(user) || user.incapacitated())
+		return FALSE
+
+	if(!can_modify)
+		if(tool)
+			to_chat(user, SPAN_WARNING("\The [holder]'s power supply cannot be removed."))
+			return TRUE // Tool interactions should get a warning, inhand interactions should just default to regular attack_hand.
+		return FALSE
+
+	// Check existing cell.
+	var/obj/item/cell/existing_cell = loaded_cell_ref?.resolve()
+	if(!istype(existing_cell) || QDELETED(existing_cell) || existing_cell.loc != holder)
+		to_chat(user, SPAN_WARNING("\The [holder] has no cell loaded."))
+		if(loaded_cell_ref)
+			loaded_cell_ref = null
+			holder_item.update_icon()
+		return TRUE // technically a valid interaction.
+
+	// Apply tool checks.
+	if(requires_tool)
+		// No tool provided means we're probably using an empty hand - don't print a warning if it's a tool based removal.
+		if(!istype(tool) || QDELETED(tool))
+			return FALSE
+		if(!IS_TOOL(tool, requires_tool))
+			var/decl/tool_archetype/tool_arch = GET_DECL(requires_tool)
+			to_chat(user, SPAN_WARNING("\The [holder] requires \a [tool_arch.name] to remove the cell."))
+			return TRUE // technically a valid interaction.
+		if(unload_delay && !tool.do_tool_interaction(requires_tool, user, holder, unload_delay))
+			return FALSE
+	else if(istype(tool))
+		return FALSE
+	// Apply general delay.
+	else if(unload_delay && !do_after(user, unload_delay, holder))
+		return FALSE
+
+	// Recheck inputs.
+	if(!istype(user) || QDELETED(user) || user.incapacitated())
+		return FALSE
+	// Recheck existing cell.
+	existing_cell = loaded_cell_ref?.resolve()
+	if(!istype(existing_cell) || QDELETED(existing_cell) || existing_cell.loc != holder)
+		to_chat(user, SPAN_WARNING("\The [holder] has no cell loaded."))
+		if(loaded_cell_ref)
+			loaded_cell_ref = null
+			holder_item.update_icon()
+		return TRUE // technically a valid interaction.
+
+	// Unload the cell.
+	user.visible_message(SPAN_NOTICE("\The [user] removes \the [existing_cell] from \the [holder]."))
```

Comment:
```suggestion
	user.visible_message("<b>\The [user]</b> removes \the [existing_cell] from \the [holder].", SPAN_NOTICE("You remove \the [existing_cell] from \the [holder]."))
```

Reply:
technically at some point this should use `SPAN_NAME()` or something, or some kind of terrible macro like `#define FORMAT_LIKE_EMOTE(user, message) SPAN_GAME(SPAN_NAME("user") + SPAN_MESSAGE("message")`.

or you could just keep it as-is and use SPAN_NOTICE